docs/core/seeding.rst

```diff
@@ -263,18 +321,30 @@ compatible with both the current middle SP and current bottom SP.
 
 .. code-block:: cpp
 
-   // A and B allow calculation of impact params in U/V plane with linear
-   // function
-   // (in contrast to having to solve a quadratic function in x/y plane)
-   float Im = std::abs((A - B * rM) * rM);
-
-   if (Im <= m_config.impactMax) {
-     topSpVec.push_back(compatTopSP[t]);
-     // inverse diameter is signed depending if the curvature is
-     // positive/negative in phi
-     curvatures.push_back(B / std::sqrt(S2));
-     impactParameters.push_back(Im);
-   }
+	// A and B allow calculation of impact params in U/V plane with linear
+	// function
+	// (in contrast to having to solve a quadratic function in x/y plane)
+	float Im;
+	if (m_config.useDetailedDoubleMeasurementInfo == false) {
+		Im = std::abs((A - B * rM) * rM);
+	} else {
+		Im = std::abs((A - B * rMxy) * rMxy);
+	}
+
+	if (Im <= m_config.impactMax) {
+		state.topSpVec.push_back(state.compatTopSP[t]);
+		// inverse diameter is signed depending if the curvature is
+		// positive/negative in phi
+		state.curvatures.push_back(B / std::sqrt(S2));
+		state.impactParameters.push_back(Im);
+
+		// evaluate eta and pT of the seed
+		float cotThetaAvg = std::sqrt(cotThetaAvg2);
+		float theta = std::atan(1. / cotThetaAvg);
+		float eta = -std::log(std::tan(0.5 * theta));
+		state.etaVec.push_back(eta);
+		state.ptVec.push_back(pT);
+	}
 
 The bottom SP and middle SP as well as the collection of top SP is passed to
```

Comment:
Sorry, can't understand this. Maybe simply start with saying that the seeds are filtered after being crated. And instead of details of function calls explain why and how?